app/controllers/api/instances_controller.rb

```diff
@@ -140,8 +140,12 @@ def pause_instance(instance)
     end
 
     def validate_instance_for_action(instance, action)
-      validation = instance.send("validate_#{action}")
-      action_result(validation[:available], validation[:message].to_s)
+      if instance.respond_to?("supports_#{action}?")
+        action_result(instance.public_send("supports_#{action}?"), instance.unsupported_reason(action.to_sym))
```

Comment:
Is there an existing API for features, or is it possible to enhance it for use cases such as these? i.e. If we have to use it with `respond_to?`/`public_send` that seems like a deficiency. @durandom any thoughts?